src/NHibernate/Bytecode/Lightweight/ReflectionOptimizer.cs

```diff
@@ -163,10 +163,10 @@ private static void EmitCastToReference(ILGenerator il, System.Type type)
 
 		private GetPropertyValueInvoker GenerateGetPropertyValueMethod(IGetter getter)
 		{
+			if (getter == null)
+				return null;
```

Comment:
I forgot the `null` case... Now that we do no more yield just null when the argument is not optimizable, the case it is null has to be handled separately, because the null coalesce operator does not work on method groups. (And the ternary requires some explicit casting.)